src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/CreateGithubIssueTest.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Octokit;
+using Xunit;
+
+namespace DependencyUpdateErrorProcessor.Tests
+{
+    public class CreateGithubIssueTest : DependencyUpdateErrorProcessorTests
+    {
+        [Theory]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "ProcessPendingUpdatesAsync", "no arguments" , "2200/1/1" , false)]
+        [InlineData("https://github.com/maestro-auth-test/maestro-test2", "38", "UpdateAssetsAsync", "[\"ee8cdcfb-ee51-4bf3-55d3-08d79538f94d\",\"UpdateAssetsAsync\"]",  "2200/1/1", false)]
+        public async Task ShouldCreateIssue(string repoUrl, string branch , string method, string arguments , DateTime errorOccurredAt, bool success)
+        {
+            Mock<RepositoryBranchUpdateHistoryEntry> repositoryBranchUpdate =
+                new Mock<RepositoryBranchUpdateHistoryEntry>();
+            repositoryBranchUpdate.Object.Repository = repoUrl;
+            repositoryBranchUpdate.Object.Branch = branch;
+            repositoryBranchUpdate.Object.Method = method;
+            repositoryBranchUpdate.Object.Timestamp = errorOccurredAt;
+            repositoryBranchUpdate.Object.Arguments = arguments;
+            repositoryBranchUpdate.Object.Success = success;
+            Mock<Octokit.Repository> repository = new Mock<Repository>();
+            GithubClient.Setup(x => x.Repository.Get(It.IsAny<string>(), It.IsAny<string>())).ReturnsAsync(repository.Object);
```

Comment:
Since they are all messing with the shared mock, it's probably better if they are in different files... technically this should be fine for XUnit, but people have been changing a lot of things to NUnit (since it has better parallelism control) and this could cause weird flakiness later as they all fight over the mock. :-(

Reply:
@ChadNedzlek  need some help in understanding this please. 

Shared mock here is github client, so you are proposing to move that to a different file?

Reply:
I added a comment to take care of shared mocks if we port these tests to Nunit.